superset/dashboards/filters.py

```diff
@@ -138,3 +140,22 @@ def apply(self, query: Query, value: Any) -> Query:
         )
 
         return query
+
+
+class FilterRelatedRoles(BaseFilter):
+    """
+    A filter to allow searching for related roles of a resource.
+
+    Use in the api by adding something like:
+    related_field_filters = {
+      "roles": RelatedFieldFilter("name", FilterRelatedRoles),
+    }
+    """
+
+    name = _("Role")
+    arg_name = "roles"
+
+    def apply(self, query: Query, value: Optional[Any]) -> Query:
+        role_model = security_manager.role_model
+        like_value = "%" + cast(str, value) + "%"
+        return query.filter((role_model.name).ilike(like_value),)
```

Comment:
instead of casting, don't we want to just return all roles if they apply? Something like
```python
if value:
    return query.filter(role_model.name.ilike(f'%{value}%'),)
return query
```
Then we could also discard the `cast`.

Reply:
sure 
adding tests also